javalib/src/main/scala/java/net/InetAddress.scala

```diff
@@ -38,8 +38,10 @@ import scala.scalanative.meta.LinktimeInfo.{isLinux, isMac}
  *    That information is handled separately.
  */
 
-class InetAddress protected (ipAddress: Array[Byte], originalHost: String)
-    extends Serializable {
+class InetAddress protected (
+    private[net] var ipAddress: Array[Byte],
```

Comment:
Sorry, I do not understand why `var` is added. Seems to have nothing to do with UDP and increases visibility.

Reply:
This is required to implement `def peek(i: InetAddress): Int`
> Peek at the packet to see who it is from. Updates the specified InetAddress to the address which the packet came from.

I turned the `ipAddress` bytes as `var` so the address can be mutated by the function. I didn't find another way around that.

Reply:
```
“Curiouser and curiouser!” Cried Alice  
  -- Lewis Carroll, Alice's Adventures in Wonderland / Through the Looking-Glass
```

Please let me submit an alternate implementation for consideration.   I have been much influenced
by Scala's emphasis on immutable data structures.  Of course, when arrays are involved, definitions
of "immutable" become squishy.

At this point, we should ask small children, those of sensitive disposition, and anyone sensible
to leave the room, cover their ears, and avert their eyes.

My suggestion is to let complexity live where it is created and limits it corrosive influence.
That is, I would keep the prior non-var declaration of the variable itself and add a short 
design note 
```
 private[net] var ipAddress: Array[Byte], // Beware contents can change, see Note 1 below
```
Then in "Note 1" or such, I would explain that almost all methods do not alter the contents
of the `ipAddress` array, but that `java.net.DatagramSocketImpl#peek` (and peekData??) 
do change contents of the array.

"I'm not bad, I am just specified that way" `DatagramSocketImpl#peek` and ilk take an 
`InetAddress` argument. `InetAddress` has (only) two direct known subclasses:  `Inet4Address`
and `Inet6Address`.  The former should hold an Array of size 4; the latter of size 16.
If the array sizes are not as expected, throw.    One could get more paranoid and 
check, probably once, if IPv4 is forced (there is a method, I do not remember the name).
If yes, then buffer size should always be 4. In no, then 16. Any other values are errors.

Now the method first compares the sizes of the "from" and "argument" addresses.  If they
are not the same, throw an exception (Yes, there are some things which could be done
to be more friendly here. I think they all boil down to "Something is not right here, so throw!"
but we could do more tracing.)

If they _are_ the same size, and do not have the same contents, then "from" can be copied
into "argument" Array.   (I am taking the bet here that the amortized time spent of "always check &
occasionally write" will be better on larger cache machines than "always write".)
In a better world, this could be experimentally determined.